stan/math/prim/fun/singular_values.hpp

```diff
@@ -17,7 +17,8 @@ namespace math {
  * @param m Specified matrix.
  * @return Singular values of the matrix.
  */
-template <typename EigMat, require_eigen_matrix_dynamic_t<EigMat>* = nullptr>
+template <typename EigMat, require_eigen_matrix_dynamic_t<EigMat>* = nullptr,
+          require_not_vt_var<EigMat>* = nullptr>
```

Comment:
It looks like this returns matrices of size zero instead of throwing. We'll need to make this consistent across all the functions but let's wait for a response to the question in `prim/fun/svd_U.hpp` first.

Reply:
Just make the prim version throw I think. That'll be simplest for now. We can roll it back. Later.

So instead of returning a 0x0, do: `check_nonzero_size("singular_values", "m", m);`